lib/globus/client/endpoint.rb

```diff
@@ -61,22 +57,33 @@ def connection
         )
       end
 
+      # Builds up a path from a list of path elements. E.g., input would look like:
+      #     ["mjgiarlo", "work123", "version1"]
+      # And this method returns:
+      #     ["/uploads/mjgiarlo/", "/uploads/mjgiarlo/work123/", "/uploads/mjgiarlo/work123/version1/"]
+      def paths
+        path_segments.map.with_index do |_segment, index|
+          File.join(config.uploads_directory, path_segments.slice(..index)).concat("/")
+        end
+      end
+
+      def path_segments
+        [user_id, "work#{work_id}", "version#{work_version}"]
+      end
+
       def endpoint
         "/v0.10/operation/endpoint/#{config.transfer_endpoint_id}"
       end
 
       # @return [Faraday::Response]
       def call_mkdir(path)
-        response = connection.post("#{endpoint}/mkdir") do |req|
+        connection.post("#{endpoint}/mkdir") do |req|
           req.headers["Content-Type"] = "application/json"
           req.body = {
             DATA_TYPE: "mkdir",
             path:
           }.to_json
         end
-        UnexpectedResponse.call(response) unless response.success?
```

Comment:
The `UnexpectedResponse.call` here meant that the calls up in `#mkdir` were dead code, and there's some smarts up there, so I figured what we really wanted was the logic up in `#mkdir`.